### ISSUE TITLE
error LGHT0094 : Unresolved reference to symbol 'File:Registrator.exe'

### ISSUE THREAD

**Issue by KlausWilly** [oleg-shilo/wixsharp#290]
Hello together;

I have a small Problem testing Wixsharp.
I test on Win10/64 
When i try to build the Project _Samples\AllInOne\MyApp.cs_ then i get 
the Error described as Header

Do you know what is the reaason for ?

thanks in advance 
Klaus

**Comment by oleg-shilo** [comment]
OK, I was wrong. It is not a defect but rather a weak sample. 
The first argument in `new InstalledFileAction("Registrator.exe",` is the id of the `File` with path `@"AppFiles\Registrator.exe"`. This id was hard-codded in the sample and only worked if the id was allocated according the old legacy algorithm that was in place at the time when the sample was written.

Today an improved algorithm is in place so the sample needs to be updated. It's done: https://github.com/oleg-shilo/wixsharp/commit/06b663c9fd5794aaa3c81567462a82d8e4fcf586.

Basically you need to use explicit ID to ensure custom actions being bound to the correct file:

```C#
new File(new Id("myapp_exe"), binaries, @"AppFiles\MyApp.exe",
new File(new Id("registrator_exe"), binaries, @"AppFiles\Registrator.exe"),
...
new InstalledFileAction("registrator_exe", "", Return.check, When.After, Step.InstallExecute, Condition.NOT_Installed),
new InstalledFileAction("registrator_exe", "/u", Return.check, When.Before, Step.InstallExecute, Condition.Installed),
new InstalledFileAction("myapp_exe", "")
```

**Comment by KlausWilly** [comment]
Hi Oleg-shilo

Thanks for correcting. I will do more testing next Days.

It's a cool thing you have done !! (-:)

have a nice day, Klaus

**Comment by oleg-shilo** [comment]
Thank you Klaus.
